cfgmgr/vxlanmgr.cpp

```diff
@@ -421,9 +630,109 @@ bool VxlanMgr::doVxlanTunnelMapDeleteTask(const KeyOpFieldsValuesTuple & t)
 
     std::string vxlanTunnelMapName = kfvKey(t);
     std::replace(vxlanTunnelMapName.begin(), vxlanTunnelMapName.end(), config_db_key_delimiter, delimiter);
-    m_appVxlanTunnelMapTable.del(vxlanTunnelMapName);
 
-    SWSS_LOG_NOTICE("Delete vxlan tunnel map %s", vxlanTunnelMapName.c_str());
+    delAppDBTunnelMapTable(vxlanTunnelMapName);
+
+    SWSS_LOG_INFO("Delete vxlan tunnel map %s", vxlanTunnelMapName.c_str());
+
+  // ip link del dev {{VXLAN}}
+    size_t found = vxlanTunnelMapName.find(delimiter);
+    const auto vxlanTunnelName = vxlanTunnelMapName.substr(0, found);
+
+    std::string vxlan_dev_name,vlan,vni_id;
+    MapCacheT map_entry;
+
+    try
+    {
+      map_entry = m_vxlanTunnelMapCache.at(vxlanTunnelMapName);
+    }
+    catch (const std::out_of_range& oor)
+    {
+      SWSS_LOG_ERROR("Error deleting tunnmap : %s exception : %s", 
+                      vxlanTunnelMapName.c_str(), oor.what());
+      return true;
+    }
+
+    vxlan_dev_name = map_entry.vxlan_dev_name;
+    vlan = map_entry.vlan;
+    vni_id = map_entry.vni_id;
+    deleteVxlanNetdevice(vxlan_dev_name);
+
+    m_vxlanTunnelMapCache.erase(vxlanTunnelMapName);
+    m_vlanMapCache.erase(vlan);
+    m_vniMapCache.erase(vni_id);
+    m_vxlanTunnelCache[vxlanTunnelName].vlan_vni_refcnt--;
+
+    //Delete the state table map of vlan to tunnel name.
+    std::string vlan_delimiter = "-";
+    found = vxlan_dev_name.find(vlan_delimiter);
+    std::string key = "Vlan" + vxlan_dev_name.substr(found+1,vxlan_dev_name.length());
+    SWSS_LOG_INFO("Delete Tunnel Map for %s -> %s ", key.c_str(), vxlan_dev_name.c_str());
+    m_stateTunnelVlanMapTable.del(key);
+    return true;
+}
+
+bool VxlanMgr::doVxlanEvpnNvoCreateTask(const KeyOpFieldsValuesTuple & t)
+{
+    SWSS_LOG_ENTER();
+
+    std::string EvpnNvoName = kfvKey(t);
+
+    if(m_EvpnNvoCache.find(EvpnNvoName) != m_EvpnNvoCache.end())
+    {
+      SWSS_LOG_ERROR("Only Single NVO object allowed");
+      return true;
+    }
+    
+    for (auto i : kfvFieldsValues(t))
+    {
+        const std::string & field = fvField(i);
+        const std::string & value = fvValue(i);
+        //if(m_vxlanTunnelCache.find(value) == m_vxlanTunnelCache.end())
+        if(!isTunnelActive(value))
+        {
+           SWSS_LOG_ERROR("NVO %s creation failed. VTEP not present",EvpnNvoName.c_str());
+           return false;
+        }
+        if (field == SOURCE_VTEP)
```

Comment:
miss to `#define SOURCE_VTEP "source_vtep"`

Reply:
will add macros for all missing